homeassistant/components/wemo/fan.py

```diff
@@ -322,6 +328,9 @@ def set_speed(self, speed: str) -> None:
             )
             self._available = False
 
+        self._update()
+        self.async_write_ha_state()
```

Comment:
You can't call async methods from sync context. Call `self.schedule_update_ha_state()` instead.